components/problem_tab.py

```diff
@@ -0,0 +1,255 @@
+    kwargs['use_cqm_solver'] = 'Constrained Quadratic Model' == kwargs['solver']
+    if kwargs['input_type'] == 'Random':
+        data = generate_data(**kwargs)
+    elif kwargs['input_type'] == 'Random Cut':
+        data = random_cut_generator(**kwargs)
+    else:
+        data = read_instance(kwargs['data_filepath'])
+    path = f'input/{file_name(kwargs)}.json'
+    result = main(data=data, solution_file=path, **kwargs)
+    if result['feasible']:
+        result['figure'].update_layout(
+            margin=dict(l=20, r=20, t=20, b=20),
+        )
+        kwargs['solution'] = result['solution']
+        return dcc.Graph(figure=result['figure'], style={'height': '700px'}), \
+               "Feasible Solution Found", 'success', kwargs
+    elif result['suitable']:
+        return '', \
+               "No Feasible Solution Found", 'danger', kwargs
+    else:
+        return '', "MIP cannot support quadratic interactions", 'danger', kwargs
+
+
+@app.callback(
+    [Output(f"div_{setting['id']}", 'style') for setting in settings[1:]],
+    Input('input_type', 'value'),
+    prevent_initial_callback=True
+)
+def update_inputs(value):
+    if 'Random' in value:
+        return displays
+    return [{'display': 'none' if d['display'] == 'block' or ab else 'block'}
+            for ab, d in zip(always_blocked, displays)]
+
+
+@app.callback(
+    Output("time_limit", "value"),
+    [Input("time_limit", "value"), Input("solver", "value")],
+
+)
+def adjust_time_limit(tl, solver):
+    if tl < 5 and solver == 'Constrained Quadratic Model':
+        return 5
+    return dash.no_update
+
+
+def generate_data(num_bins, num_cases,
+                  bin_length, bin_width, bin_height,
+                  case_size_range_min,
+                  case_size_range_max, seed, **kwargs):
+    rng = np.random.default_rng(seed)
+    data = {
+        "num_bins": num_bins,
+        "bin_dimensions": [bin_length, bin_width, bin_height],
+        "case_length": rng.integers(
+            case_size_range_min, case_size_range_max,
+            num_cases, endpoint=True
+        ),
+        "case_width": rng.integers(
+            case_size_range_min, case_size_range_max,
+            num_cases, endpoint=True
+        ),
+        "case_height": rng.integers(
+            case_size_range_min, case_size_range_max,
+            num_cases, endpoint=True
+        ),
+    }
+    case_dimensions = np.vstack(
+        [data["case_length"], data["case_width"], data["case_height"]]
+    )
+    unique_dimensions, data["quantity"] = np.unique(case_dimensions,
+                                                    axis=1,
+                                                    return_counts=True)
+
+    data["case_length"] = unique_dimensions[0, :]
+    data["case_width"] = unique_dimensions[1, :]
+    data["case_height"] = unique_dimensions[2, :]
+
+    data["case_ids"] = np.array(range(len(data["quantity"])))
+    return data
+
+
+def file_name(p):
+    from itertools import chain
```

Comment:
Can't you import this at the top of the file?

Reply:
Of course, I can. Laziness...